jcl/src/java.base/share/classes/com/ibm/oti/reflect/TypeAnnotationParser.java

```diff
@@ -109,7 +125,17 @@
 	 */
 	public  static AnnotatedType buildAnnotatedSupertype(Class clazz) {
 		byte[] attr = getAttributeData(clazz);
-		AnnotatedType annotatedSuperclass = sun.reflect.annotation.TypeAnnotationParser.buildAnnotatedSuperclass(attr, AnnotationParser.getConstantPool(clazz), clazz);
+		long offset = Unsafe.ARRAY_BYTE_BASE_OFFSET + ((attr.length * Unsafe.ARRAY_BYTE_INDEX_SCALE) - VM.FJ9OBJECT_SIZE);
+		long ramCPAddr = 0;
+		if (VM.FJ9OBJECT_SIZE == 4) {
+			/* Compressed object refs */
+			ramCPAddr = Unsafe.getUnsafe().getInt(attr, offset);
```

Comment:
here you should use `Integer.toUnsignedLong(Unsafe.getUnsafe().getInt(attr, offset));`

I believe this is the cause for the sign extension. You'll need this change in a couple other places as well.